src/main/java/games/strategy/engine/framework/AbstractGame.java

```diff
@@ -40,7 +41,7 @@
   protected IGameModifiedChannel m_gameModifiedChannel;
   protected final PlayerManager m_playerManager;
   protected boolean m_firstRun = true;
-  protected final ListenerList<GameStepListener> m_gameStepListeners = new ListenerList<>();
+  protected final List<GameStepListener> gameStepListeners = new CopyOnWriteArrayList<>();
```

Comment:
This class doesn't look like it's serialized but did you check for reflective access?  I got burned once in the RMI area because there is some utility class that assumes class fields are always prefixed with `m_`.  Not sure if `IGame`s are affected or not.  You might want to try to play a network game between this branch and `master` to verify.